File: src/test/java/com/naminhyeok/fantazzk/architecture/ModulithStructureTest.java
package com.naminhyeok.fantazzk.architecture;

import static org.assertj.core.api.Assertions.assertThatCode;

import com.naminhyeok.fantazzk.FantazzkApplication;
import org.junit.jupiter.api.Test;
import org.springframework.modulith.core.ApplicationModules;

class ModulithStructureTest {
    @Test
    void 현재_자바_구조에서_모듈_검증이_통과한다() {
        assertThatCode(() -> ApplicationModules.of(FantazzkApplication.class).verify())
            .doesNotThrowAnyException();
    }
}

File: application-api/src/test/kotlin/com/naminhyeok/fantazzk/architecture/ModulithStructureTest.kt
package com.naminhyeok.fantazzk.architecture

import com.naminhyeok.fantazzk.bootstrap.root.FantazzkApplication
import org.assertj.core.api.Assertions.assertThatCode
import org.junit.jupiter.api.Test
import org.springframework.modulith.core.ApplicationModules

class ModulithStructureTest {
    @Test
    fun `room과 template 모듈 구성이 검증된다`() {
        assertThatCode {
            ApplicationModules.of(FantazzkApplication::class.java).verify()
        }.doesNotThrowAnyException()
    }
}
